### ISSUE TITLE
Check number of required user index searches required to display webpages.

### ISSUE THREAD

**Issue by phochste** [LibreCat/LibreCat#629]
In the `dev` branch `3+ 6*NUM_OF_PUBLICATIONS` are required to load the **My Dashboard** page. Is this a bug or feature and can it be optimised?

**Comment by vpeil** [comment]
I can't see how this can be a feature!

**Comment by phochste** [comment]
The source of the requests are LibreCat::App::Catalogue::Controller::Permission::Permissions functions  `can_edit` , `can_download` etc..that in the templates only receive the `user_id` and need to query the database for every link.

There was in the past a question how much can be cached of these requests, there was no agreement on caching. There was also a question why the `user_id` gets pased around in the templates and not the complete user record (which is not possible because the templates now are targeted to passing record fields around.. and not the record hash itself).

**Comment by phochste** [comment]
E.g. currently the templates are being executed like:

```
template('file.tt' , %$record)
```

which leaves no room to extend the number of fields in the template. This would be better:

```
template('file.tt' , record =>$record)
```

But this means editing all templates.

The caching of searches was not seen as very helpful because it is hard to control when user records are changed, how to invalidate the cache.

**Comment by vpeil** [comment]
Well, get rid of **this particular problem** one needs to change 2 - 3 templates. The rest of the templates can be changed successiveley.

**Comment by vpeil** [comment]
Some benchmarking: dropping the three lines with the `can_*` calls works 20x faster!

I think we can drop the `can_edit` condition, because if you see the record you can edit it anyway. So this check is not needed (except: in some future use case one wants to display stuff but make it uneditable)

**Comment by phochste** [comment]
I'm more in favour of some sort of caching, as these quite detailed control in who can view, edit, etc is required in some of my use cases. We can discuss this in our workshop?

**Comment by vpeil** [comment]
yes, let's discuss it in our workshop!

**Comment by ThriceGood** [comment]
has there been any code changes to it lately to try and address this situation? Because since I recently updated my system to the latest dev branch I've noticed a huge increase in the performance compared how it was when I first raised this issue in the slack channel. I think its as performant as it was before.

**Comment by vpeil** [comment]
@phochste hasn't this issue been solved by PR #879 ?

**Comment by phochste** [comment]
@vpeil it has been solved for now indeed
